doc/tools/apigen.py

```diff
@@ -457,9 +457,24 @@ def write_index(self, outdir, froot='gen', relative_to=None):
         w = idx.write
         w('.. AUTO-GENERATED FILE -- DO NOT EDIT!\n\n')
 
-        title = "API Reference"
+        title = "API Reference for scikit-image version 0.12.3"
         w(title + "\n")
         w("=" * len(title) + "\n\n")
+
+        for f in self.written_modules:
+            module_name = f.split('.')
```

Comment:
Let's add a comment here to show what's happening:

```
# We look at the module name.  If it is `skimage`, display, if `skimage.submodule`, only show `submodule`,
# if it is `skimage.submodule.subsubmodule`, ignore.
```

Also, let's change the logic below slightly, to something like this:

```
if len(module_name) > 2:
    continue
elif len(module_name) == 1:
    display_name = module_name[0]
elif len(module_name) == 2:
    display_name = module_name[1]

w('- `{0} <{1}.html`__\n'.format(os.path.join(display_name, module_name)))
```